vta/include/vta/driver.h

```diff
@@ -119,20 +119,25 @@ void VTAMemFree(void* buf);
 vta_phy_addr_t VTAMemGetPhyAddr(void* buf);
 
 /*!
- * \brief Flushes the region of memory out of the CPU cache to DRAM.
- * \param buf Pointer to memory region allocated with VTAMemAlloc to be flushed.
- *            This need to be the physical address.
- * \param size Size of the region to flush in Bytes.
+ * \brief Performs a copy operation from host memory to buffer allocated with VTAMemAlloc.
+ * \param dst The desination buffer in FPGA-readable memory. Has to be allocated with VTAMemAlloc().
+ * \param src The source buffer in host memory.
+ * \param size Size of the region in Bytes.
+ * \param flush When the memory is shared between the FPGA and host, trigger cache flush
+ *              before the transer to make the data visible to the FPGA.
  */
-void VTAFlushCache(vta_phy_addr_t buf, int size);
+void VTAMemMoveToBuffer(void* dst, const void* src, size_t size, bool flush);
```

Comment:
flush and invalidate should be an implicit property instead of an argument?

Reply:
consider unify the API similar to the runtime API VTABufferCopy

Reply:
I was a little conflicted about where to insert it but I think what you suggest makes more sense. I'll go ahead and make the change.

Reply:
What makes me a little conflicted is the fact that if we want to present a general API for all FPGA-based systems, cache flushing and invalidation is only specific to shared memory systems. Therefore I thought to hide it behind the `VTAMemMoveToBuffer`/`VTAMemMoveFromBuffer` API. In addition, since it can be absorbed under these functions, I don't think it needs to ever be exposed in the driver API.

Reply:
I've addressed your comments for now; let me know if you are happy with the current implementation